.github/workflows/cmake.yml

```diff
@@ -0,0 +1,62 @@
+name: "CMake Release"
+
+on:
+  [push, workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+            - name: Windows Tiles Sounds x64 MSVC
+              preset: windows-tiles-sounds-x64-msvc
+              runs-on: windows-latest
+            - name: Windows Tiles Sounds x64
+              preset: windows-tiles-sounds-x64
+              runs-on: ubuntu-22.04
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.runs-on }}
+    env:
+      DevEnvDir: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/
+    steps:
+      - name: Update
+        if: ${{ matrix.runs-on == 'ubuntu-22.04' }}
+        run: |
+            sudo apt-get update
+            sudo apt-get install ninja-build mingw-w64 gettext
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+      - uses: actions/cache@v3
+        with:
+            # C:/Users/runneradmin/AppData/Local/vcpkg/archives
+          path: |
+            ${{ github.workspace }}/out/build/${{ matrix.preset }}
+            ${{ env.LOCALAPPDATA }}/vcpkg/archives
+            /home/runner/.cache/vcpkg/archives
+            ${{ env.VCPKG_INSTALLATION_ROOT }}/downloads
+            ${{ env.VCPKG_INSTALLATION_ROOT }}/buildtrees
+            ${{ env.VCPKG_INSTALLATION_ROOT }}/installed
+            ${{ env.VCPKG_INSTALLATION_ROOT }}/packages
+          key: ${{ matrix.preset }}/${{ github.ref }}
+          restore-keys: |
+              ${{ matrix.preset }}
```

Comment:
1) Can we use [@lukka/run-vcpkg](https://github.com/lukka/run-vcpkg) which should already handle cmake integration and auto caching? Or is there something it's not doing which is needed?
2) We definitely *don't* want to cache `buildtrees` because that's unnecessary intermediates that'll choke up the cache space, which we don't have overhead for..
3) If we do end up using run-vcpkg, we should wait until the author pushes a fix for https://github.com/lukka/run-vcpkg/issues/155 so it can use zstd instead of gzip on windows.
4) I'm not sure we have the cache space to afford another whole configuration, especially once we add ccache support to the MSVC build in https://github.com/CleverRaven/Cataclysm-DDA/pull/60385 (which is blocked on point 2 above as well). That's going to eat about a gigabyte all told and we're up to ~4.35GB currently at my last math, so I'm already going to have to trim out some cache from the other builds to fit.

Reply:
Removing GHA with interactive rebasing.
Will review all of this in a future PR, thanks.